editor/editor_node.cpp

```diff
@@ -3985,6 +3985,8 @@ void EditorNode::_scene_tab_changed(int p_tab) {
 	if (p_tab == editor_data.get_edited_scene())
 		return; //pointless
 
+	editor_data.set_edited_scene(p_tab);
```

Comment:
I'm not sure this is the proper fix, as:
- It's redundant with what `set_current_scene` will do
- It's not included in the `UndoRedo` stack, so it can't be undone